client/signup/utils.js

```diff
@@ -17,16 +17,18 @@ var i18nUtils = require( 'lib/i18n-utils' ),
 	formState = require( 'lib/form-state' );
 
 function getFlowName( parameters ) {
-	var currentFlowName = flows.currentFlowName;
-	if ( parameters.flowName && isFlowName( parameters.flowName ) ) {
-		return parameters.flowName;
-	}
+	const flow = ( parameters.flowName && isFlowName( parameters.flowName ) ) ? parameters.flowName : defaultFlowName;
```

Comment:
I'm ok with assuming that `filterFlowName` is a function (although I do like to code defensively), but I don't like the idea of allowing it to return any value without validating the result. Since it's defined in what amounts to a config file (`flows.js`), it would be very easy for it to accidentally return null or `''` or something incorrect and break all signup flows.